pub/helpers/bin/infer_sdk_versions.dart

```diff
@@ -95,33 +95,53 @@ Future<void> main(List<String> args) async {
 String get flutterReleasesUrl =>
     'https://storage.googleapis.com/flutter_infra_release/releases/releases_linux.json';
 
-// Retrieves all released versions of Flutter.
+// Retrieves all released versions of Flutter, applying a 30-day cooldown filter when possible.
 Future<List<FlutterRelease>> retrieveFlutterReleases(String url) async {
   final response = await client.get(Uri.parse(url));
   final decoded = jsonDecode(response.body);
   if (decoded is! Map) throw FormatException('Bad response - should be a Map');
   final releases = decoded['releases'];
   if (releases is! List)
     throw FormatException('Bad response - releases should be a list.');
+
   final result = <FlutterRelease>[];
+  final now = DateTime.now().toUtc();
+  final cooldownDays = Duration(days: 30);
```

Comment:
can we use the cooldown feature in private beta instead? I know we already have this for python.